packages/cli/openapi-parser/src/openapi/v3/extensions/fernExtensions.ts

```diff
@@ -408,7 +408,31 @@ export const FernOpenAPIExtension = {
      *     - "2.0"
      *     - "Latest"
      */
-    FERN_VERSION: "x-fern-version"
+    FERN_VERSION: "x-fern-version",
+
+    /**
+     * Allows users to specify the encoding of the type. For example, suppose you need to configure
+     * Protobuf-encoding details like the following:
+     *
+     * User:
+     *  properties:
+     *    username:
+     *      type: string
+     *  x-fern-encoding:
+     *    proto:
+     *      type: user.v1.User
+     */
+    ENCODING: "x-fern-encoding",
+
+    /**
+     * Allows users to configure gRPC services. This must be specified on individual service
+     * declarations.
+     *
+     * x-fern-transport:
+     *   grpc:
+     *     service-name: UserService
+     */
+    TRANSPORT: "x-fern-transport"
```

Comment:
Don't see any logic that actually pushes thes extensions into the fern definition. is that intentional?

Reply:
Yes, nothing implemented just yet - just spec'ing out the interface for now. I'll incorporate this when they're used in a follow-up.